include/ruby/buffer.h

```diff
@@ -0,0 +1,108 @@
+#ifndef RUBY_BUFFER_H
```

Comment:
`#pragma once`

Reply:
I decided that we cannot use `#pragma once` here due to the following comment: https://github.com/ruby/ruby/blob/master/include/ruby/internal/arithmetic.h#L16-L20

```
 * @note       To  ruby-core:  remember  that   this  header  can  be  possibly
 *             recursively included  from extension  libraries written  in C++.
 *             Do not  expect for  instance `__VA_ARGS__` is  always available.
 *             We assume C99  for ruby itself but we don't  assume languages of
 *             extension libraries. They could be written in C++98.
```

I don't know `#pragma once` can be used with any C++98 compilers.

Reply:
@kou @mrkn Take a look at https://github.com/ruby/ruby/pull/3023 for `#pragma once` discussions.

Reply:
It’s very informative!
(Old Sun C!)